src/main/java/org/opensearch/knn/index/codec/BasePerFieldKnnVectorsFormat.java

```diff
@@ -121,8 +123,22 @@ public KnnVectorsFormat getKnnVectorsFormatForField(final String field) {
             return vectorsFormatSupplier.apply(knnVectorsFormatParams);
         }
 
+        boolean buildGraph = getBuildGraphSetting();
         // All native engines to use NativeEngines990KnnVectorsFormat
-        return new NativeEngines990KnnVectorsFormat(new Lucene99FlatVectorsFormat(FlatVectorScorerUtil.getLucene99FlatVectorsScorer()));
+        return new NativeEngines990KnnVectorsFormat(
+            new Lucene99FlatVectorsFormat(FlatVectorScorerUtil.getLucene99FlatVectorsScorer()),
+            buildGraph
+        );
+    }
+
+    private boolean getBuildGraphSetting() {
+        // Default should be true, only explicit false should disable building graph for
+        // native engine
+        if (mapperService.isEmpty()) {
+            return true;
+        }
```

Comment:
if MapperService is empty the code should have crashed in line 77. I think we should get the mapperservice from line 77 and pass it in the function.